### PR TITLE
DEV-2752: Allow nullable agency abbreviations

### PR DESCRIPTION
**Pull request by kbard**
**High level description:**

Agency abbreviations are nullable.  Tweaks to prevent the word "null" from showing up when nulls are provided in the response.

**JIRA Ticket:**

[DEV-2752](https://federal-spending-transparency.atlassian.net/browse/DEV-2752)

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] API contract already has these fields marked as nullable
- [x] There is no dependency on back end code
- [x] Verified cross-browser compatibility

### CODE REVIEW

**Comment by MaxwellKendall** [comment]
@kbard 

[This line](https://github.com/fedspendingtransparency/usaspending-website/pull/1273/files#diff-a21af102d6e27e78f7b6a0124fff60e6R50) appears to be redundant based on the other updates.

**Comment by MaxwellKendall** [comment]
Also would be nice to update the relevant tests to ensure undefined values translate to `''`

**Comment by kbard** [comment]
> @kbard
> 
> [This line](https://github.com/fedspendingtransparency/usaspending-website/pull/1273/files#diff-a21af102d6e27e78f7b6a0124fff60e6R50) appears to be redundant based on the other updates.

Believe it or not, this code isn't run through a model.

File: src/js/containers/search/filters/AgencyListContainer.jsx [hunk 1]
```diff
@@ -47,10 +47,10 @@ export default class AgencyListContainer extends React.Component {
             results.forEach((item) => {
                 let subAbbreviation = '';
                 let topAbbreviation = '';
-                if (item.subtier_agency.abbreviation !== '') {
+                if (item.subtier_agency.abbreviation !== '' && item.subtier_agency.abbreviation !== null) {
```
Comment (MaxwellKendall) [inline src/js/containers/search/filters/AgencyListContainer.jsx]: Isn't this null check unnecessary given our null check in the models?

Also, we could just do a falsey check as follows:

```javascript
if (!obj.property) `
```
which should handle these two conditions in one expression.

File: src/js/containers/search/filters/AgencyListContainer.jsx [hunk 2]
```diff
@@ -47,10 +47,10 @@ export default class AgencyListContainer extends React.Component {
...
                     subAbbreviation = `(${item.subtier_agency.abbreviation})`;
                 }
-                if (item.toptier_agency.abbreviation !== '') {
+                if (item.toptier_agency.abbreviation !== '' && item.toptier_agency.abbreviation !== null) {
                     topAbbreviation = `(${item.toptier_agency.abbreviation})`;
```
Comment (MaxwellKendall) [inline src/js/containers/search/filters/AgencyListContainer.jsx]: Ditto ☝️

File: src/js/containers/search/filters/AgencyListContainer.jsx (hunk 1 above)
Comment (kbard) [reply src/js/containers/search/filters/AgencyListContainer.jsx]: > Believe it or not, this code isn't run through a model.

I will make it falsy.  I was just following some code I saw elsewhere.

File: tests/containers/search/filters/agencyFilter/AgencyListContainer-test.jsx [hunk 3]
```diff
@@ -149,6 +147,18 @@ describe('AgencyListContainer', () => {
             expect(container.state().autocompleteAgencies[2].title).toEqual('DEF Agency (DEF)');
             expect(container.state().autocompleteAgencies[2].subtitle).toEqual('Sub-Agency of Department ABC (ABC)');
         });
+        it('should not contain agency abbreviations', async () => {
```
Comment (MaxwellKendall) [inline tests/containers/search/filters/agencyFilter/AgencyListContainer-test.jsx]: Shouldn't this say "should not contain _null_ agency abbreviations"?

File: tests/containers/search/filters/agencyFilter/AgencyListContainer-test.jsx [hunk 4]
```diff
@@ -149,6 +147,18 @@ describe('AgencyListContainer', () => {
...
+            const container = setupShallow(initialFilters);
+            container.setState({
+                agencySearchString: 'abc'
+            });
+
+            container.instance().parseAutocompleteAgencies(mockNullAgencyResults);
+
+            // Toptier agencies should have been moved to the top of the list
```
Comment (MaxwellKendall) [inline tests/containers/search/filters/agencyFilter/AgencyListContainer-test.jsx]: Let's also update this comment which is a duplicate of `line 144`

Comment (kbard) [reply tests/containers/search/filters/agencyFilter/AgencyListContainer-test.jsx]: oops

**Pull request merged**